src/js/bun/sql.ts

```diff
@@ -468,42 +520,29 @@ class ConnectionPool {
         this.onAllQueriesFinished();
       }
     }
+
     if (connection.state !== PooledConnectionState.connected) {
       // connection is not ready
-      return;
-    }
-    if (was_reserved) {
-      if (this.waitingQueue.length > 0) {
-        if (connection.storedError) {
-          // this connection got a error but maybe we can wait for another
-
-          if (this.hasConnectionsAvailable()) {
-            return;
-          }
+      if (connection.storedError) {
+        // this connection got a error but maybe we can wait for another
 
-          // we have no connections available so lets fails
-          let pending;
-          while ((pending = this.waitingQueue.shift())) {
-            pending.onConnected(connection.storedError, connection);
-          }
+        if (this.hasConnectionsAvailable()) {
           return;
         }
-        const pendingReserved = this.reservedQueue.shift();
-        if (pendingReserved) {
-          connection.flags |= PooledConnectionFlags.reserved;
-          connection.queryCount++;
-          // we have a connection waiting for a reserved connection lets prioritize it
-          pendingReserved(connection.storedError, connection);
-          return;
+
+        // we have no connections available so lets fails
+        let pending;
+        while ((pending = this.waitingQueue.shift())) {
```

Comment:
Is it intentional that this is .shift instead of a for loop? Should we swap the arrays and then for loop?